src/foraging_gui/Foraging.py

```diff
@@ -2119,11 +2119,20 @@ def closeEvent(self, event):
         # stop the current session first
         self._StopCurrentSession() 
 
+        reply = None
         if self.unsaved_data:
-            reply = QMessageBox.critical(self, 
-                'Box {}, Foraging Close'.format(self.box_letter), 
+            reply = QMessageBox.critical(self,
+                'Box {}, Foraging Close'.format(self.box_letter),
                 'Exit without saving?',
-                QMessageBox.Yes | QMessageBox.No , QMessageBox.No)  
+                QMessageBox.Yes | QMessageBox.No , QMessageBox.No)
+            if reply == QMessageBox.No:
+                event.ignore()
+                return
+        elif self.WeightAfter.text() == '' and self.StartANewSession == 1 and reply != QMessageBox.Yes and not self.unsaved_data:  # post weight not entered and session ran
```

Comment:
How would reply have been set by this point?

Reply:
I was trying to avoid the situation where if user said it was fine that there was unsaved data , the weight pop up wouldn't show since the user already said it was fine. I think that's taken care of though by saying not self.unsaved_data so I can remove the reply != QMessagebox.Yes